pkg/entity-store/store.go

```diff
@@ -136,43 +151,117 @@ func (e *BaseEntity) setModifiedTime(modifiedTime time.Time) {
 	e.ModifiedTime = modifiedTime
 }
 
+func (e *BaseEntity) setRevision(revision uint64) {
+	e.Revision = revision
+}
+func (e *BaseEntity) setVersion(version uint64) {
+	e.Version = version
+}
+func (e *BaseEntity) setSpec(spec Spec) {
+	e.Spec = spec
+}
+
 func (e *BaseEntity) setStatus(status Status) {
 	e.Status = status
 }
 
-func (e *BaseEntity) setReason(reason []string) {
+func (e *BaseEntity) setReason(reason Reason) {
 	e.Reason = reason
 }
 
-func (e *BaseEntity) GetModifiedTime() time.Time {
-	return e.ModifiedTime
+func (e *BaseEntity) setTags(tags Tags) {
+	e.Tags = tags
+}
+func (e *BaseEntity) setDelete(delete bool) {
+	e.Delete = delete
+}
+
+// getKey builds the key for a give entity
+func (e *BaseEntity) getKey(dt dataType) string {
+	return buildKey(e.OrganizationID, dt, e.Name)
+}
+
+func (e *BaseEntity) GetID() string {
+	return e.ID
 }
 
 // GetName retreives the entity name
 func (e *BaseEntity) GetName() string {
 	return e.Name
 }
 
-// GetTags retreives the entity tags
-func (e *BaseEntity) GetTags() Tags {
-	return e.Tags
+func (e *BaseEntity) GetOrganizationID() string {
+	return e.OrganizationID
 }
 
-// getKey builds the key for a give entity
-func (e *BaseEntity) getKey(dt dataType) string {
-	return buildKey(e.OrganizationID, dt, e.Name)
+func (e *BaseEntity) GetCreateTime() time.Time {
+	return e.CreatedTime
+}
+
+func (e *BaseEntity) GetModifiedTime() time.Time {
+	return e.ModifiedTime
+}
+
+func (e *BaseEntity) GetRevision() uint64 {
+	return e.Revision
+}
+
+func (e *BaseEntity) GetVersion() uint64 {
+	return e.Version
 }
 
 func (e *BaseEntity) GetStatus() Status {
 	return e.Status
 }
 
+func (e *BaseEntity) GetDelete() bool {
+	return e.Delete
+}
+
+func (e *BaseEntity) GetReason() Reason {
+	return e.Reason
+}
+func (e *BaseEntity) GetSpec() Spec {
+	return e.Spec
+}
+
+// GetTags retreives the entity tags
+func (e *BaseEntity) GetTags() Tags {
+	return e.Tags
+}
+
 type entityStore struct {
```

Comment:
I think I was a bit confused because we should probably rename this to KVEntityStore or something more descriptive.

Reply:
Yeah, I want to do that, too.